src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -803,14 +863,388 @@ def forward(
         if audio_codes is not None and audio_scales is None:
             raise ValueError("You specified `audio_codes` but did not specify the `audio_scales`")
 
-        if audio_scales is not None and audio_codes is None:
-            raise ValueError("You specified `audio_scales` but did not specify the `audio_codes`")
+        if audio_codes is None:
+            audio_codes, audio_scales = self.encode(input_values, padding_mask, bandwidth, return_dict=False)
+
+        decoded_output = self.decode(audio_codes, audio_scales)
+        audio_values = decoded_output.audio_values[:, :, : input_values.shape[-1]]
 
-        if audio_scales is None and audio_codes is None:
-            audio_codes, audio_scales = self.encode(input_values, padding_mask, bandwidth, False)
+        outputs = {
+            "audio_codes": audio_codes,
+            "audio_values": audio_values,
+        }
+
+        if self.training and self.loss_function is not None:
+            reconstruction_loss, commitment_loss = self.loss_function(
+                model=self, input_values=input_values, audio_values=audio_values
+            )
+            outputs["reconstruction_loss"] = reconstruction_loss
+            outputs["commitment_loss"] = commitment_loss
 
-        audio_values = self.decode(audio_codes, audio_scales, padding_mask, return_dict=return_dict)[0]
         if not return_dict:
-            return (audio_codes, audio_values)
+            return tuple(outputs.values())
+
+        return EncodecOutput(**outputs)
+
+
+"""
+    Discriminator code copied over and refactored from https://github.com/facebookresearch/encodec/blob/main/encodec/msstftd.py#L28
+"""
```

Comment:
You don't need to add this here, it's already assumed the code is adapted from it